bs/src/tablecloth.mli

```diff
@@ -1527,41 +1527,345 @@ module Result : sig
+  valid values (`'Some`), and one for invalid values (`'None`).
+*)
+
 module Option : sig
+  (**
+    `type` is the type constructor for an `Option` type. You specify
+    the type of the `Some` variant.
+    
+    ### Example
+    
+    Here is how you would annotate an `Option` variable whose `Some`
+    variant is an integer:
+    
+    ```ocaml
+    let x: (int) Tablecloth.Option.t = Some 3
+    let y: (int) Tablecloth.Option.t = None
+    ```
+    
+    ```reason
+    let x: Tablecloth.Result.t(string, int) = Ok(3);
+    let y: Tablecloth.Result.t(string, int) = Error("bad");
+    ```
+  *)
   type 'a t = 'a option
 
+  (**
+    `andThen(~f = fcn, opt)` applies function `fcn`, which takes a non-`Option`
+    parameter and returns a `Option`, to an `Option` variable `opt`.
+    
+    If `opt` is of the form `Some(x)`, `andThen` returns `f(x)`;
+    otherwise `andThen` returns `None`.
+    
+    (Same as `and_then`.)
+    
+    ### Example
+    ```reason
+    let recip = (x: float): Tablecloth.Option.t(float) => {
+      if (x == 0.0) {
+        None;
+      } else {
+        Some(1.0 /. x);
+      }
+    };
+    
+    andThen(~f = recip, Some(4.0)) == Some(0.25);
+    andThen(~f = recip, None) == None;
+    andThen(~f = recip, Some(0.0)) == None;
+    ```
+    
+    `andThen` is usually used to implement a chain of function
+    calls, each of which returns an `Option` value.
+    
+    ```reason
+    let root = (x: float): Tablecloth.Option.t(float) => {
+      if (x < 0.0) {
+        None
+      } else {
+        Some(sqrt(x));
+      }
+    };
+    
+    root(4.0) |> andThen(~f = recip) == Some(0.5);
+    root(-2.0) |> andThen(~f = recip) == None;
+    root(0.0) |> andThen(~f = recip) == None;
+    ```
+  *)
   val andThen : f:('a -> 'b option) -> 'a option -> 'b option
 
+  (**
+    `and_then ~f:fcn opt` applies function `fcn`, which takes a non-`Option`
+    parameter and returns an `Option`, to an `Option` variable `opt`.
+    
+    If `opt` is of the form `Some x`, `and_then` returns `f x`;
+    otherwise it returns `None`.
+    
+    (Same as `andThen`.)
+    
+    ### Example
+    ```ocaml
+    let recip (x:float) : float Tablecloth.Option.t = (
+      if (x == 0.0) then
+        None
+      else
+        Some (1.0 /. x)
+    )
+    
+    and_then ~f:recip (Some 4.0) = Some 0.25
+    and_then ~f:recip None = None
+    and_then ~f:recip (Some 0.0) = None
+    ```
+    
+    `and_then` is usually used to implement a chain of function
+    calls, each of which returns an `Option` value.
+    
+    ```ocaml
+    let root (x:float) : float Tablecloth.Option.t = (
+      if (x < 0.0) then
+        None
+      else
+        Some (sqrt x)
+    )
+    
+    root 4.0 |> and_then ~f:recip = Some 0.5
+    root (-2.0) |> and_then ~f:recip = None 
+    root(0.0) |> and_then ~f:recip = None
+    ```
+  *)
   val and_then : f:('a -> 'b option) -> 'a option -> 'b option
 
+  (**
+    `or_ opt_a opt_b` (`or_(opt_a, opt_b)` in ReasonML) returns
```

Comment:
Might want to point out that this doesn't short-circuit.

Reply:
I am not sure of this wording:

Note that `or_` does not short-circuit; when you call `or_`, both arguments are evaluated before being passed to the function.

How does that sound to you?

Reply:
Not sure how to word this. How does this sound:

Unlike the built in `or` operator, the `or_` function does not short-circuit. When you call the function, both the arguments are evaluated so they can be passed to the function.

Reply:
Oops. For some reason, I didn't see my comment, so I duplicated it. Let me know which version of the wording you prefer.

Reply:
Either one is fine, thanks!

Reply:
Done (after next commit/push)